cranelift/codegen/src/ir/jumptable.rs

```diff
@@ -23,42 +27,59 @@ pub struct JumpTableData {
 
 impl JumpTableData {
     /// Create a new empty jump table.
-    pub fn new() -> Self {
-        Self { table: Vec::new() }
+    pub fn new(def: Block) -> Self {
+        Self { table: vec![def] }
     }
 
-    /// Create a new empty jump table with the specified capacity.
-    pub fn with_capacity(capacity: usize) -> Self {
-        Self {
-            table: Vec::with_capacity(capacity),
-        }
-    }
     /// Create a new jump table with the provided blocks
-    pub fn with_blocks(table: Vec<Block>) -> Self {
+    pub fn with_blocks(def: Block, mut table: Vec<Block>) -> Self {
+        table.push(def);
         Self { table }
     }
 
+    /// Fetch the default block for this jump table.
+    pub fn default_block(&self) -> Block {
+        *self.table.last().unwrap()
+    }
+
     /// Get the number of table entries.
     pub fn len(&self) -> usize {
-        self.table.len()
+        self.table.len() - 1
     }
 
     /// Append a table entry.
     pub fn push_entry(&mut self, dest: Block) {
-        self.table.push(dest)
+        let last = self.table.len();
+        self.table.push(dest);
+
+        // Ensure that the default block stays as the final element in the table.
```

Comment:
Maybe keep it as first element instead?

Reply:
The only downside there is that we would break the invariant that the table data is 0-indexed. However, as we can always return a slice to the table data, maybe that's fine?

Reply:
I think so.